lib/services/swagger-types-mapper.ts

```diff
@@ -13,7 +13,7 @@ export class SwaggerTypesMapper {
       const { type } = param as ParamWithTypeMetadata;
       const typeName =
         type && isFunction(type)
-          ? this.mapTypeToOpenAPIType(type.name)
+          ? this.mapTypeToOpenAPIType(type().name)
```

Comment:
```
type && isFunction(type)
```
will give `true` for type references (class = function) too. We can't simply evaluate this